include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -0,0 +1,221 @@
+    static inline OutputIterator push_interval(Longitude const& lon1,
+                                               Longitude const& lon2,
+                                               OutputIterator oit)
+    {
+        typedef longitude_interval<Longitude> interval_type;
+
+        typedef math::detail::constants_on_spheroid
+            <
+                Longitude, radian
+            > constants;
+
+        BOOST_ASSERT(! math::larger(lon1, lon2));
+        BOOST_ASSERT(! math::larger(lon1, constants::max_longitude()));
+
+        if (math::larger(lon2, constants::max_longitude()))
+        {
+            *oit++ = interval_type(lon1, constants::max_longitude());
+            *oit++ = interval_type(constants::min_longitude(),
+                                   lon2 - constants::period());
+        }
+        else
+        {
+            *oit++ = interval_type(lon1, lon2);
+        }
+        return oit;
+    }
+
+    template <typename Linear, typename Box>
+    static inline void apply(Linear const& linear, Box& mbr)
+    {
+        typedef typename coordinate_type<Box>::type box_coordinate_type;
+        typedef typename fp_coordinate_type<Linear>::type calculation_type;
+
+        typedef typename geometry::segment_iterator
+            <
+                Linear const
+            > iterator_type;
+
+        typedef typename coordinate_system
+            <
+                Linear
+            >::type::units linear_units_type;
+
+        BOOST_ASSERT(geometry::num_points(linear) != 0);
+
+        typedef longitude_interval<calculation_type> interval_type;
+
+        calculation_type const pi = math::pi<calculation_type>();
+
+        std::vector<interval_type> longitude_intervals;
+        std::back_insert_iterator
+            <
+                std::vector<interval_type>
+            > oit(longitude_intervals);
+
+        calculation_type lat_min = 0, lat_max = 0;
+        bool first = true;
+        for (iterator_type seg_it = geometry::segments_begin(linear);
+             seg_it != geometry::segments_end(linear);
+             ++seg_it, first = false)
+        {
+            typename point_type<Linear>::type p[2];
+            detail::assign_point_from_index<0>(*seg_it, p[0]);
+            detail::assign_point_from_index<1>(*seg_it, p[1]);
+
+            calculation_type lon1 = geometry::get<0>(p[0]);
+            calculation_type lat1 = geometry::get<1>(p[0]);
+            calculation_type lon2 = geometry::get<0>(p[1]);
+            calculation_type lat2 = geometry::get<1>(p[1]);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon1, lat1);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    linear_units_type
+                >(lon2, lat2);
+
+            math::convert_coordinates<linear_units_type, radian>(lon1, lat1);
+            math::convert_coordinates<linear_units_type, radian>(lon2, lat2);
```

Comment:
Indeed it should work better in the case of integral degrees, but AFAIU only in this case. For FP degrees normalized and converted to FP radians it shouldn't change anything.

My point is, it probably isn't possible to represent true 90 as it certainly isn't possible to represent true pi/2. And while converting between both units we can loose information. It's just a guess but in some cases, for some input in FP degrees or some user-defined type the value after conversion could be outside the range e.g. [-pi/2, pi/2] for latitude. I just think it'd be safer to normalize exactly those values that can be passed into some trigonometric functions. The radians are exactly those values. Maybe the radians should also be normalized (2nd time) or maybe only the integral degrees should be normalized at the beginning (and FP radians always normalized at the end)?